### DISCUSSION TITLE
Validation of multiple dependent fields

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#4494] [Q&A]
Hi all, I have multiple (unlimited) date ranges the user can add. At the end, I want to validate all given ranges and see if there are any gaps. As an example:

Defined range:
- 2021/01/01 - 2021/12/31

User ranges:
- 2021/01/01 - 2021/05/01
- 2021/07/01 - 2021/12/31

=> Gap 2021/05/01 - 2021/07/01

Currently I am checking this using my own validation method:

```javascript
const isGap = (definedRange, userRanges) => () => {
  ...
}
```
```javascript
rules: {{
  validation: {
    isGap: isGap({...}, [{...}, {...}])
  }
}}
```

But if I attach this method to every range field, I generate the error everywhere, which I don't like. Also the revalidation does not work correctly (I have to manually change every range field to revalidate and reset the error). Is there a better solution for this? Thanks!

**Comment by bluebill1049** [comment]
use `getValues()` to retrive other fields value

**Reply by ghost** [reply]
Thank you @bluebill1049 . I get the other field values using the `useWatch` hook. I have written a curried validation function which takes the needed field values and checks the date ranges. I attached this validation method to all fields (from, to) using the `rules` prop on the controller. This validation now sets all fields to error at once. If I then adjust the date range and close the gap, all other untouched fields using the custom validation remain on error...
